### PR TITLE
Updates Package Requirements

### PR DESCRIPTION
**Pull request by jeff-hernandez**
The following changes were made:
- python requirement for Read the Docs was updated to 3.6
- dev requirements were consolidated into the test requirements
- codecov version was updated to 2.1.0

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,5 +1,11 @@
-fastparquet>=0.1.6
 pytest==4.4.1
-pytest-xdist==1.26.1
 pytest-cov==2.6.1
+fastparquet>=0.1.6
+codecov==2.1.0
+flake8==3.7.8
+autopep8==1.4.4
+isort==4.3.21
+jupyter==1.0.0
```
Comment (rwedge) [inline test-requirements.txt:8]: Do the test cases use jupyter?

Comment (jeff-hernandez) [reply test-requirements.txt:8]: No, will remove.

**Comment by jeff-hernandez** [comment]
Hmm, not sure why pandas can't find `fastparquet` when we remove `nbsphinx` from the test requirements. Any idea why this is the case?

**Comment by rwedge** [comment]
> Hmm, not sure why pandas can't find `fastparquet` when we remove `nbsphinx` from the test requirements. Any idea why this is the case?

Looks like it's because `nbsphinx` installs the `packaging` module, which is an unlisted dependency for fastparquet 0.4.0

https://github.com/dask/fastparquet/issues/501

**Comment by jeff-hernandez** [comment]
I see. Do you think we should replace `nbsphinx` with `packaging` or keep as is?

**Comment by rwedge** [comment]
Let's replace with packaging, and when new version of fastparquet is released we can drop the extra dependency

**Pull request merged**